### PR TITLE
Add plugin for huajiao.com and zhanqi.tv

### PR DESCRIPTION
**Pull request by fozzysec**
plugin for zhanti.tv from https://github.com/steven7851/livestreamer/blob/patch-3/src/livestreamer/plugins/zhanqi.py

### CODE REVIEW

File: src/streamlink/plugins/zhanqi.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
+import re
+
+from streamlink.plugin import Plugin
+from streamlink.plugin.api import http, validate
+from streamlink.plugin.api.utils import parse_json
+from streamlink.stream import (
+    HTTPStream, HLSStream
+)
+
+API_URL = "http://www.zhanqi.tv/api/public/room.liveparam?room_id={roomID[id]}"
```
Comment (steven7851) [inline src/streamlink/plugins/zhanqi.py]:
```python
API_URL = "http://www.zhanqi.tv/api/static/live.roomid/{roomID[id]}.json"
```

File: src/streamlink/plugins/zhanqi.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+STATUS_ONLINE = 4
+STATUS_OFFLINE = 0
+
+_url_re = re.compile("""
+    http(s)?://(www\.)?zhanqi.tv
+    /(?P<channel>[^/]+)
+""", re.VERBOSE)
+
+_json_re = re.compile(r"window\.oPageConfig\.oRoom\s=\s({.+?});")
```
Comment (steven7851) [inline src/streamlink/plugins/zhanqi.py]:
```python
_json_re = re.compile(r"window\.oPageConfig\.oRoom\s*=\s*({.+?});")
```

File: src/streamlink/plugins/zhanqi.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        if not room:
+            return
+
+        if room["status"] != STATUS_ONLINE:
+            return
+
+        hls_url = "http://dlhls.cdn.zhanqi.tv/zqlive/{room[videoId]}_1024/index.m3u8?Dnion_vsnae={room[videoId]}".format(room=room)
+        hls_stream = HLSStream(self.session, hls_url)
+        if hls_stream:
+            yield "hls", hls_stream
```
Comment (steven7851) [inline src/streamlink/plugins/zhanqi.py]: This HLS url is sometimes not working.

Comment (fozzysec) [inline src/streamlink/plugins/zhanqi.py]: I will update the PR, thanks.

File: src/streamlink/plugins/zhanqi.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+        roomID = http.get(self.url, schema=_roomID_schema)
+
+        res = http.get(API_URL.format(roomID=roomID))
+        room = http.json(res, schema=_room_schema)
+        if not room:
+            return
+
+        if room["status"] != STATUS_ONLINE:
+            return
```
Comment (beardypig) [inline src/streamlink/plugins/zhanqi.py]: Maybe a `self.logger.info` message to say that the stream is not currently available?

**Comment by fozzysec** [comment]
**zhanqi.py**:
`STREAM_WEIGHTS` added, and according to above review about hls source's problem, `HTTPStream` is preferred.

**Comment by fozzysec** [comment]
I think this will works with `--stream-types` and `--stream-priority`

`fozzy@Laptop ~ streamlink http://www.zhanqi.tv/swxg --stream-types="hls,http,rtmp" --stream-priority="hls" 
[cli][info] Found matching plugin zhanqi for URL http://www.zhanqi.tv/swxg
[cli][info] Available streams: hls (best, worst)
[cli][info] Opening stream: hls (hls)
`
`fozzy@Laptop ~ streamlink http://www.zhanqi.tv/swxg --stream-types="hls,http,rtmp" --stream-priority="http"
[cli][info] Found matching plugin zhanqi for URL http://www.zhanqi.tv/swxg
[cli][info] Available streams: http (best, worst)
[cli][info] Opening stream: http (http)`

**Comment by beardypig** [comment]
Looks good to me :)

**Comment by gravyboat** [comment]
Thanks @fozzysec!

**Pull request merged**